common/meson.build

```
@@ -0,0 +1,89 @@
+install_headers('pkcs11.h', 'pkcs11x.h', subdir: 'p11-kit-1/p11-kit')
+
+libp11_common_sources = [
+  'argv.c', 'argv.h',
+  'attrs.c', 'attrs.h',
+  'array.c', 'array.h',
+  'buffer.c', 'buffer.h',
+  'compat.c', 'compat.h',
+  'constants.c', 'constants.h',
+  'debug.c', 'debug.h',
+  'dict.c', 'dict.h',
+  'hash.c', 'hash.h',
+  'lexer.c', 'lexer.h',
+  'message.c', 'message.h',
+  'path.c', 'path.h',
+  'pkcs11.h', 'pkcs11x.h', 'pkcs11i.h',
+  'runtime.c', 'runtime.h',
+  'url.c', 'url.h',
+  'init.h'
+]
+
+libp11_common = static_library('libp11-common.a', libp11_common_sources,
+                               include_directories: [configinc, commoninc])
+
+libp11_library_sources = [
+  'library.c', 'library.h'
+]
+
+libp11_library = static_library('libp11-library.a', libp11_library_sources,
```

There is no need to include `library.h` because the compiler will found it, so only `library.c` is needed. The fact that it's only one file, why don't you include directly in the `static_library` call? It will avoid the creation/memory of `libp11_library_sources` variable.

```suggestion
libp11_library = static_library('libp11-library.a', 'library.c',
```